### PR TITLE
Issue#67 pluginclient api

### PR DESCRIPTION
**Pull request by GrandSchtroumpf**
**Ready to merge**
Fix issue https://github.com/ethereum/remix-plugin/issues/67

### CODE REVIEW

File: src/store/base.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
+// BASE
+export class BaseStore<S extends {}> {
```
Comment (yann300) [inline src/store/base.ts]: seems not used anymore - shall we remove it?

Comment (GrandSchtroumpf) [reply src/store/base.ts]: Yes right.

File: src/client/iframe.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+          event.source.postMessage(message, event.origin as any)
+        })
+      }
+
+      // Check if is loaded
+      if (!loaded) throw new Error('Handshake before communicating')
+
+      switch (action) {
+        case 'notification': {
+          client.events.emit(`[${name}] ${key}`, ...payload)
```
Comment (yann300) [inline src/client/iframe.ts]: interesting way of providing separation

Comment (yann300) [reply src/client/iframe.ts]: indeed

File: src/client/iframe.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    "https://remix.ethereum.org",
+    ...localhost
+  ].includes(msgOrigin)
+}
+
+/**
+ * Start listening on the IDE though PostMessage
+ * @param store A store to put the messages into
+ */
+function connectIframe(client: PluginClient) {
```
Comment (yann300) [inline src/client/iframe.ts]: as a first tentative, we could try to implement function `connectWS(client: PluginClient) {`

File: src/client/iframe.ts (hunk 2 above)
Comment (yann300) [reply src/client/iframe.ts]: should that one use https://github.com/ethereum/remix-plugin/pull/68/files#diff-2609dc589ff34d954b51407ff2140828R32 ?

**Pull request merged**